plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -168,5 +168,14 @@ func defaultPriorities() sets.String {
 				Weight: 1,
 			},
 		),
+		factory.RegisterPriorityConfigFactory(
+			"NodePreferAvoidPriority",
+			factory.PriorityConfigFactory{
+				Function: func(args factory.PluginFactoryArgs) algorithm.PriorityFunction {
+					return priorities.NewNodePreferAvoidPriority(args.ControllerLister)
+				},
+				Weight: 1,
```

Comment:
You should give this a high weight, because when a pod's controller is in PreferAvoidPods, we want that node to be used only as a last choice. It's not easy to choose a good weight because it depends on the other weights (and in fact today the total priority can overflow, see #24720 which I just filed). I would suggest setting this to 10000 and adding this comment above it:
```
// Set this weight large enough to override all other priority functions.
// TODO: Figure out a better way to do this, maybe at same time as fixing #24720.
```